priv/vmq_webhooks.schema

```diff
@@ -0,0 +1,80 @@
+                                                                                 on_client_offline,
+                                                                                 on_client_gone]}},
+                                                                    {commented, "auth_on_register"},
+                                                                    {include_default, "webhook1"}
+                                                                   ]}.
+
+%% @doc Associate an endpoint with a name.
+{mapping, "vmq_webhooks.$name.endpoint", "vmq_webhooks.user_webhooks", [
+                                                                        {datatype, string},
+                                                                        {commented, "http://localhost/myendpoints"},
+                                                                        {include_default, "webhook1"}
+                                                                       ]}.
+
+%% @doc Control if the payload should be base64 encoded.
+{mapping, "vmq_webhooks.$name.base64encode", "vmq_webhooks.user_webhooks", [
+                                                                            {datatype, flag},
+                                                                            {default, on},
+                                                                            {commented, "false"},
+                                                                            {include_default, "webhook1"},
+                                                                            hidden
+                                                                           ]}.
+
+{translation,
+ "vmq_webhooks.user_webhooks",
+ fun(Conf) ->
+         Suffixes = ["hook", "endpoint", "base64encode"],
+         UNames =
+             [
+              proplists:get_all_values(
+                "$name",
+                cuttlefish_variable:fuzzy_matches(["vmq_webhooks", "$name", Suffix], Conf))
+               || Suffix <- Suffixes ],
+         HookNames = lists:usort(lists:append(UNames)),
+         Webhooks =
+             [ {SortName,
+                #{
+                  hook => cuttlefish:conf_get("vmq_webhooks." ++ SortName ++ ".hook", Conf, undefined),
+                  endpoint => cuttlefish:conf_get("vmq_webhooks." ++ SortName ++ ".endpoint", Conf, undefined),
+                  options => 
+                      #{ base64_payload =>
+                             cuttlefish:conf_get("vmq_webhooks." ++ SortName ++ ".base64encode", Conf, true)
+                       }
+                 }}
+               || SortName <- HookNames ],
+         %% I know of no other way to abort the translation phase and
+         %% report an error back to the user than using throw. Though
+         %% it's not too pretty I think it's nicer than just warning
+         %% that something is configured incorrectly.
```

Comment:
I guess we could do something similar in other cases too, e.g. if someone configures a listener for an occupied port.